### PR TITLE
query by custom object value, close #999, close #1010

### LINKED ISSUE #999: custom object query by value field

### LINKED ISSUE #1010: in depth query model for values of custom objects

### PR DESCRIPTION
**Pull request by schleichardt**
currently the API does not support values in the query. so later I need to retrigger the tests

### CODE REVIEW

**Comment by schleichardt** [comment]
@lauraluiz please review

later I wan't to provide an in depth query model, but for now I just want to test the feature for Yann.

I also don't wan't to put it yet on the release notes.

**Comment by schleichardt** [comment]
I am done and I would add the release notes:

```html
 <li class=new-in-release>{@link CustomObject}s can be queried by their value 
with {@link CustomObjectQueryModel#value()}</li>
```

File: commercetools-models/src/main/java/io/sphere/sdk/customobjects/queries/CustomObjectValueRootJsonQueryModelImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+final class CustomObjectValueRootJsonQueryModelImpl<T> extends QueryModelImpl<T> implements RootJsonQueryModel<T>, JsonQueryModel<T>, JsonValueQueryModel<T> {
+
+    public static final String VALUE = "value";
+
+    CustomObjectValueRootJsonQueryModelImpl(@Nullable final QueryModel<T> parent, @Nullable final String pathSegment) {
+        super(parent, pathSegment);
+    }
+
+    @Override
+    public JsonQueryModel<T> ofObject() {
```
Comment (lauraluiz) [inline commercetools-models/src/main/java/io/sphere/sdk/customobjects/queries/CustomObjectValueRootJsonQueryModelImpl.java]: For this and `ofValue`, better rename with `Json` prefix

**Comment by lauraluiz** [comment]
Besides the docu, :+1:

**Comment by lauraluiz** [comment]
Docu looks good :+1:

**Pull request merged**